src/main/java/sparksoniq/spark/ml/RumbleMLUtils.java

```diff
@@ -174,7 +174,7 @@ private static Object convertRumbleAtomicToJava(AtomicItem atomicItem, String ja
             case "double":
                 return atomicItem.castAs(AtomicTypes.DoubleItem).getDoubleValue();
             case "long":
-                return (long) atomicItem.castAs(AtomicTypes.DoubleItem).getDoubleValue();
+                return atomicItem.castAs(AtomicTypes.DecimalItem).getDecimalValue().longValue();
```

Comment:
Fixed a misconception on DecimalItem to DataType.Long conversion.

Reply:
Nice fix.